langchain4j-alloydb-pg/src/main/java/dev/langchain4j/data/document/loader/alloydb/AlloyDBLoader.java

```diff
@@ -0,0 +1,266 @@
+        StringBuilder sb = new StringBuilder();
+        for (String column : contentColumns) {
+            if (row.containsKey(column)) {
+                sb.append(row.get(column)).append(", ");
+            }
+        }
+        return sb.toString().trim().replaceAll(", $", ""); // Remove trailing comma
+    }
+
+    private static String yamlFormatter(Map<String, Object> row, List<String> contentColumns) {
+        StringBuilder sb = new StringBuilder();
+        for (String column : contentColumns) {
+            if (row.containsKey(column)) {
+                sb.append(column).append(": ").append(row.get(column)).append("\n");
+            }
+        }
+        return sb.toString().trim();
+    }
+
+    private static String jsonFormatter(Map<String, Object> row, List<String> contentColumns) {
+        JsonObject json = new JsonObject();
+        for (String column : contentColumns) {
+            if (row.containsKey(column)) {
+                json.addProperty(column, (String) row.get(column));
+            }
+        }
+        return json.toString();
+    }
+
+    /**
+     * Loads data from Alloy Db in form of {@code Document}.
+     *
+     * @return List<Document> list of documents
+     * @throws SQLException if databse error occurs
+     */
+    public List<Document> load() throws SQLException {
+        List<Document> documents = new ArrayList<>();
+        PreparedStatement statement = engine.getConnection().prepareStatement(query);
+        ResultSet resultSet = statement.executeQuery();
+
+        while (resultSet.next()) {
+            Map<String, Object> rowData = new HashMap<>();
+            for (String column : contentColumns) {
+                rowData.put(column, resultSet.getObject(column));
+            }
+            for (String column : metadataColumns) {
+                rowData.put(column, resultSet.getObject(column));
+            }
+            if (metadataJsonColumn != null) {
+                rowData.put(metadataJsonColumn, resultSet.getObject(metadataJsonColumn));
+            }
+
+            Document doc = parseDocFromRow(rowData);
+            documents.add(doc);
+        }
+
+        return documents;
+    }
+
+    private Document parseDocFromRow(Map<String, Object> row) {
+        String pageContent = formatter.apply(row, contentColumns);
+        Metadata metadata = new Metadata();
+
+        if (metadataJsonColumn != null && row.containsKey(metadataJsonColumn)) {
+            try {
+                JsonObject jsonMetadata = JsonParser.parseString(
+                        row.get(metadataJsonColumn).toString()).getAsJsonObject();
+                for (String key : jsonMetadata.keySet()) {
+                    metadata.put(key, jsonMetadata.get(key).toString());
+                }
+            } catch (JsonSyntaxException e) { // Handle JSON parsing errors
+                log.debug(e.getMessage(), e.getCause());
```

Comment:
Throw error instead of ignoring it. Add a nice error message to note the issue is unmarshalling the JSON metadata.